training/iso-builder/README.md

```diff
@@ -0,0 +1,52 @@
+Installation iso builder
+---
+
+This directory provides an easy way to build installation `iso` images with a container and a kickstart
+embedded.
+
+# Makefile targets
+
+| Target      | Description                                             |
+|-------------|---------------------------------------------------------|
+| image       | Build the container                                     |
+| iso         | Create bootable installation iso                        |
+
+# Makefile variables
+
+| Variable                  | Description                                     | Default                                     |
+|---------------------------|-------------------------------------------------|---------------------------------------------|
+| FROM                      | Overrides the base image for the Containerfile  | `quay.io/centos/centos:stream9`             |
+| REGISTRY                  | Container Registry for storing container images | `quay.io`                                   |
+| REGISTRY_ORG              | Container Registry organization 	     	      | `ai-lab`                                    |
+| IMAGE_NAME                | Container image name                            | `iso-builder`                               |
+| IMAGE_TAG                 | Container image tag                             | `latest`                                    |
+| CONTAINER_TOOL            | Container tool used for build                   | `podman`                                    |
+| CONTAINER_TOOL_EXTRA_ARGS | Container tool extra arguments                  | ` `                                         |
+| EMBED_IMAGE               | The container image to embed in the iso         | *Required*                                  |
+| ORIGINAL_ISO              | Path to the base installation iso               | *Required*                                  |
+| SSHKEY                    | The SSH public key for the root account         | *Required*                                  |
+| OUTPUT_DIR                | Path where to store the new installation iso    | `./iso`                                     |
+
+# How to build the container image
+
+The first step is simple, the `iso-builder` container image needs to be build:
+
+```
+make image
+```
+
+This is based on CentOS Stream by default, for Red Hat Enterprise Linux you would use the `FROM` variable:
+
+```
+make image FROM=registry.access.redhat.com/ubi9/ubi:latest
```

Comment:
This does not work, ubi does not ship with DNF, and will fail in the container build. Do not use this example

Reply:
Hmm.... ubi does not ship with dnf?

Reply:
Sorry my mistake, package lorax not available on ubi9:

```
 make image FROM=registry.access.redhat.com/ubi9/ubi:latest
"podman" build \
		--file Containerfile \
		--tag "quay.io/ai-lab/iso-builder:latest" \
		--from=registry.access.redhat.com/ubi9/ubi:latest \
		 \
		 \

STEP 1/5: FROM registry.access.redhat.com/ubi9/ubi:latest
STEP 2/5: RUN dnf install -y skopeo buildah lorax
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered with an entitlement server. You can use subscription-manager to register.

Red Hat Universal Base Image 9 (RPMs) - BaseOS  831 kB/s | 421 kB     00:00
Red Hat Universal Base Image 9 (RPMs) - AppStre 3.0 MB/s | 1.7 MB     00:00
Red Hat Universal Base Image 9 (RPMs) - CodeRea 379 kB/s | 114 kB     00:00
No match for argument: lorax
Error: Unable to find a match: lorax
Error: building at STEP "RUN dnf install -y skopeo buildah lorax": while running runtime: exit status 1

make: *** [image] Error 1
```

Reply:
Yeah I would probably argue that `mkksiso` should be clearly split from lorax and put in the bucket of "things we generally support in the product".  Whereas "lorax" is an implementation detail of building the anaconda ISO that can and probably will change in the future.